PhysObjectExtractor/src/MuonAnalyzer.cc

```diff
@@ -126,12 +145,30 @@ MuonAnalyzer::analyzeMuons(const edm::Event& iEvent, const edm::Handle<reco::Muo
 	  muon_eta.clear();
 	  muon_phi.clear();
 	  muon_ch.clear();
+	  muon_mass.clear();
+	  muon_pfreliso03all.clear();
+	  muon_pfreliso04all.clear();
+	  muon_tightid.clear();
+	  muon_softid.clear();
+	  muon_dxy.clear();
+	  muon_dxyErr.clear();
+	  muon_dz.clear();
+	  muon_dzErr.clear();
+	  muon_jetidx.clear();
+	  muon_genpartidx.clear();
+
+
+  Handle<reco::VertexCollection> vertices;
+  iEvent.getByLabel(InputTag("offlinePrimaryVertices"), vertices);
 
   if(muons.isValid()){
      // get the number of muons in the event
      nummuon=(*muons).size();
-        for (reco::MuonCollection::const_iterator itmuon=muons->begin(); itmuon!=muons->end(); ++itmuon){
+     const float mu_min_pt = 3;
+     math::XYZPoint pv(vertices->begin()->position());
 
+        for (reco::MuonCollection::const_iterator itmuon=muons->begin(); itmuon!=muons->end(); ++itmuon){
+          if (itmuon->pt() > mu_min_pt) {
         	    muon_e.push_back(itmuon->energy());
```

Comment:
For some reason, all these variables are not being filled, can you please check?